flyteidl/protos/flyteidl/datacatalog/datacatalog.proto

```diff
@@ -386,6 +389,8 @@ message DatasetPropertyFilter {
         string name = 2;
         string domain = 3;
         string version = 4;
+        // Optional, partition key applied to the dataset.
+        string partition = 5;
```

Comment:
`partition` -> `org`? this assumes we're storing the `org` in datacatalog i think too?

Reply:
d'oy, tysm for all the catches!